src/TypeSpec.Extension/Emitter.Csharp/package.json

```diff
@@ -34,7 +34,31 @@
     "files": [
         "dist/src/**"
     ],
+    "dependencies": {
+        "@autorest/csharp": "3.0.0-beta.20230309.1",
+        "fs-extra": "^10.1.0",
+        "js-yaml": "^4.1.0",
+        "json-serialize-refs": "^0.1.0-0",
+        "winston": "^3.8.2"
+    },
+    "peerDependencies": {
+        "@azure-tools/typespec-azure-core": ">=0.31.0 <1.0.0",
+        "@azure-tools/typespec-client-generator-core": ">=0.31.0 <1.0.0",
+        "@typespec/compiler": ">=0.45.0 <1.0.0",
+        "@typespec/http": ">=0.45.0 <1.0.0",
+        "@typespec/rest": ">=0.45.0 <1.0.0",
+        "@typespec/versioning": ">=0.45.0 <1.0.0"
+    },
     "devDependencies": {
+        "@azure-tools/typespec-azure-core": "0.31.0",
+        "@azure-tools/typespec-client-generator-core": "0.31.0",
+        "@typespec/compiler": "0.45.0",
```

Comment:
should this pr include a package-lock.json at the root level to pin all these dev dependencies?

Reply:
Good point.  I updated the `package-lock.json` that already exists in the repo root, and verified that it uses the version of `@typespec/compiler` listed in `devDependencies` (`0.45.0`), rather than `latest` (`0.45.2`).  So we shouldn't need to duplicate the `@typespec` and `@azure-tools` packages in the root `package.json`.